cdc/stats.hh

```diff
@@ -89,4 +89,33 @@ public:
     stats();
 };
 
+// Contains the details on what happened during a CDC operation.
+struct operation_details {
+    stats::part_type_set touched_parts;
+    bool was_split = false;
+    bool had_preimage = false;
+    bool had_postimage = false;
+};
+
+// This object tracks the lifetime of write handlers related to one CDC operation. After all
+// write handlers for the operation finish, CDC metrics are updated.
+struct operation_result_tracker {
```

Comment:
Maybe make it a class and drop private: label below. Also mark final.

Reply:
Done